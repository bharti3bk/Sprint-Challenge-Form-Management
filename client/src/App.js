import React from 'react';
import logo from './logo.svg';
import './App.css';
import UserRegisterationForm from './Component/RegisterationForm'

function App() {
  return (
    <div className="App">
      <h1 id = "header">Register</h1>
      <UserRegisterationForm data-testid="userForm" />
    </div>
  );
}

export default App;
