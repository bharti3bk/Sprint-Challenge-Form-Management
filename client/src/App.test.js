import React from 'react';
import {Router} from 'react-router-dom'
import App from './App';
import {render, fireEvent, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom"
import { createMemoryHistory } from 'history'
import mockAxios from 'axios'


afterEach(cleanup)
// this is a handy function that I would utilize for any component
// that relies on the router being in context
function renderWithRouter(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  }
}

it('submits the form', async () => {
  // Arrange
  const {getByPlaceholderText, getByText} = renderWithRouter(<App />)
  const userInput = getByPlaceholderText('UserName');
  userInput.nodeValue = "bharti"
  const userPassword = getByPlaceholderText('Password');
  userPassword.nodeValue = "password"

  mockAxios.post.mockImplementationOnce(() =>
  Promise.resolve({
    data: { results: {message:"user registered successfully"} }
  })
);
  // Act
  const leftClick = { button: 0 }
  fireEvent.click(getByText("Submit"), leftClick );

  // Assert
  expect(mockAxios.post).toHaveBeenCalledTimes(1);
})