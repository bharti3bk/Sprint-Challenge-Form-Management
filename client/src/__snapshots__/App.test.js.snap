// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Register
    </h1>
    <form
      class="ui form"
    >
      <div
        class="field"
      >
        <label
          id="lableOne"
        >
          UserName
        </label>
        <input
          name="username"
          placeholder="UserName"
          type="text"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <label
          id="lableThree"
        >
          Password
        </label>
        <input
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <div
          class="ui checkbox"
        >
          <input
            name="check"
            tabindex="0"
            type="checkbox"
            value=""
          />
          <label
            id="lableFour"
          >
            I agree to the Terms and Conditions
          </label>
        </div>
      </div>
      <button
        class="ui button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
